modules/serverless-config-high-workloads-serving.adoc

```diff
@@ -0,0 +1,71 @@
+// Module included in the following assemblies:
+//
+// * /knative-serving/scalability-and-performance-serving.adoc
+
+
+:_mod-docs-content-type: PROCEDURE
+[id="serverless-config-high-workloads-serving_{context}"]
+= Configuring Serving for high workloads
+
+You can configure Knative Serving for high workloads using the KnativeServing Custom Resource (CR).
+The following findings are relevant to configuring Knative Serving for a high workload:
+
+[NOTE]
+====
+These findings have been tested with requests with a payload size of 0-32 kb. The Knative Service backends used in those tests had a startup-latency between 0 to 10 seconds and response times between 0 to 5 seconds.
+====
+
+* All data-plane components are mostly increasing CPU usage on higher requests and payload scenarios, so the CPU requests and limits have to be tested and potentially increased.
+* The activator component also might need more memory, when it has to buffer more or bigger request payloads, so the memory requests and limits might need to be increased as well.
+* One activator pod can handle approximately 2500 requests per second before it starts to increase latency and, at some point, leads to errors.
+* One `3scale-kourier-gateway` or `istio-ingressgateway` pod can also handle approximately 2500 requests per second before it starts to increase latency and, at some point, leads to errors.
+* Each of the data-plane components consumes up to 1 vCPU of CPU for handling 2500 requests per second. Note that this highly depends on the payload size and the response times of the Knative Service backend.
+
+[IMPORTANT]
+====
+Fast startup and fast response-times of your Knative Service user workloads are critical for good performance of the overall system. The Knative Serving components are buffering incoming requests when the Knative Service user backend is scaling-up or when request concurrency has reached its capacity. If your Knative Service user workload introduces long startup or request-latency, it will either overload the `activator` component (when the CPU and memory configuration is too low) or lead to errors for the calling clients.
+====
+
+.Procedure
+
+* To fine-tune your installation, use the previous findings combined with your own test results to configure the `KnativeServing` custom resource:
++
+.A high workload configuration in KnativeServing CR
+[source,yaml]
+----
+apiVersion: operator.knative.dev/v1beta1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+spec:
+  high-availability:
+    replicas: 2 <1>
+  workloads:
+    - name: component-name <2>
+      replicas: 2 <2>
+      resources:
+        - container: container-name
+          requests:
+            cpu: <3>
+            memory: <3>
+          limits:
+            cpu: <3>
+            memory: <3>
```

Comment:
Duplicate callouts are not supported on docs.redhat.com, each must be a unique number. In this case, the key at the bottom is also describing multiple parameters in a single callout. It should be one-to-one.